### PR TITLE
Setup IRSA trust between kubernetes cluster and IAM for external-secrets

### PR DESCRIPTION
**Pull request by avnes**
## Checklist before requesting a review
- [x] I have tested changes in my sandbox
- [x] I have added the needed changes in the `test/integration` folder to apply my changes in QA. [Read the guide on adding environment variables in QA](https://wiki.dfds.cloud/en/ce-private/atlantis/adding-env-vars)
- [x] I have rebased the code to master (or merged in the latest from master)

## Is it a new release?
- [x] Apply a release tag `release:(major|minor|patch)`, following semantic versioning in [this guide](https://semver.org/) or `norelease` if there is no changes to the Terraform code

### CODE REVIEW

File: _sub/security/external-secrets/main.tf [hunk 1]
```diff
@@ -23,4 +23,66 @@ resource "github_repository_file" "external-secrets_helm_patch" {
...
+
+# --------------------------------------------------
+# IAM policy and role for external secrets operator
+# --------------------------------------------------
+
+# tfsec:ignore:aws-iam-no-policy-wildcards
+resource "aws_iam_policy" "this" {
+  name        = "${var.iam_role_name}-policy"
+  description = "Used by IRSA for external secrets operator"
+  policy      = <<EOF
```
Comment (aleksandra-fromm) [inline _sub/security/external-secrets/main.tf:36]: Could you please use `aws_iam_policy_document` data source instead?

File: _sub/security/external-secrets/main.tf [hunk 2]
```diff
@@ -23,4 +23,66 @@ resource "github_repository_file" "external-secrets_helm_patch" {
...
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ssm:GetParameter*",
+                "ssm:DescribeParameters",
+                "tag:GetResources"
+            ],
+            "Resource": "arn:aws:ssm:eu-west-1:${var.workload_account_id}:parameter*"
```
Comment (aleksandra-fromm) [inline _sub/security/external-secrets/main.tf:47]: I wonder if it wouldn't be better to pass the region?

Comment (avnes) [reply _sub/security/external-secrets/main.tf:47]: good call

File: _sub/security/external-secrets/main.tf (hunk 1 above)
Comment (avnes) [reply _sub/security/external-secrets/main.tf]: Done

**Pull request merged**